mantidimaging/gui/windows/spectrum_viewer/test/presenter_test.py

```diff
@@ -259,13 +259,17 @@ def test_WHEN_do_add_roi_to_table_called_THEN_roi_added_to_table(self):
         self.view.add_roi_table_row.assert_called_once_with("roi_1", (255, 0, 0))
 
     def test_WHEN_do_remove_roi_called_THEN_roi_removed(self):
-        self.presenter.model.set_new_roi("all")
-        self.presenter.view.spectrum_widget.add_roi(self.presenter.model._roi_ranges["all"], "all")
-        for _ in range(2):
-            self.presenter.do_add_roi()
-        self.assertEqual(["all", "roi", "roi_1"], list(self.presenter.model._roi_ranges.keys()))
+        self.presenter.view.spectrum_widget.roi_dict = {"all": mock.Mock(), "roi": mock.Mock(), "roi_1": mock.Mock()}
+        self.presenter.view.spectrum_widget.remove_roi = mock.Mock()
         self.presenter.do_remove_roi("roi_1")
-        self.assertEqual(["all", "roi"], list(self.presenter.model._roi_ranges.keys()))
+
+        self.presenter.view.spectrum_widget.remove_roi.assert_called_once_with("roi_1")
+        expected_roi_dict = {"all": mock.Mock(), "roi": mock.Mock()}
+        actual_roi_dict = {
+            key: value
+            for key, value in self.presenter.view.spectrum_widget.roi_dict.items() if key != "roi_1"
+        }
+        self.assertEqual(expected_roi_dict.keys(), actual_roi_dict.keys())
```

Comment:
This part is only testing if python dictionaries work, so could be removed.

All the test can usefully do is check that when `self.presenter.do_remove_roi("roi_1")` is called,  that `self.presenter.view.spectrum_widget.remove_roi()` gets called